### PR TITLE
fix pico_on_switch to set pico as CC with tests

### PR DESCRIPTION
**Pull request by gfx**
I'm not sure what `pico_on_switch()` really should do, but I'm just adding what is set as CC.

### CODE REVIEW

File: t/set_cc.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                         NULL);
+    ok(ret == 0);
+    size_t num_packets = PTLS_ELEMENTSOF(packets);
+    ret = quicly_send(client, &dest, &src, packets, &num_packets, packetsbuf, sizeof(packetsbuf));
+    ok(ret == 0);
+
+    size_t num_decoded = decode_packets(decoded, packets, num_packets);
+    ok(num_decoded == 1);
+    ret = quicly_accept(&server, &quic_ctx, NULL, &fake_address.sa, decoded, NULL, new_master_id(), NULL);
+    ok(ret == 0);
```
Comment (kazuho) [inline t/set_cc.c:44]: Considering that this is a unit test anyways, I do not think we need to actually generate packets to test the feature.

IMO, all we need to do is create a quicly_conn_t instance by calling `quicly_connect`, then set various CCs by calling `quicly_set_cc` and test them using `quicly_get_stats`.

If that works, I would appreciate making that change, then moving the test function into test.c. I do not think we want to create dedicated source files for testing each of these small features.

**Review by kazuho — COMMENTED** [review]
Thank you for the PR. The fix looks correct, and the test are encouraging :heart:

File: t/set_cc.c (hunk 1 above)
Comment (gfx) [reply t/set_cc.c]: I thought I need to test a server's connection!

* [x] use a connection created by `quicly_connect`
* [x] move `test_quicly_set_cc` to `test.c`

File: t/test.c [hunk 2]
```diff
@@ -569,6 +569,73 @@ static void test_nondecryptable_initial(void)
...
+
+    // cubic to reno
+    quicly_set_cc(conn, &quicly_cc_type_cubic);
+    quicly_set_cc(conn, &quicly_cc_type_reno);
+    ret = quicly_get_stats(conn, &stats);
+    ok(ret == 0);
+    ok(strcmp(stats.cc.type->name, "reno") == 0);
+}
+
+
```
Comment (kazuho) [inline t/test.c:638]:
```suggestion
```

File: t/test.c [hunk 3]
```diff
@@ -569,6 +569,73 @@ static void test_nondecryptable_initial(void)
 #undef LEN_LOW
 }
 
+static void test_quicly_set_cc(void)
```
Comment (kazuho) [inline t/test.c:572]:
```suggestion
static void test_set_cc(void)
```

All the tests are about quicly.

File: t/test.c [hunk 4]
```diff
@@ -642,6 +709,7 @@ int main(int argc, char **argv)
     subtest("stream-concurrency", test_stream_concurrency);
     subtest("lossy", test_lossy);
     subtest("test-nondecryptable-initial", test_nondecryptable_initial);
+    subtest("set_cc", test_quicly_set_cc);
```
Comment (kazuho) [inline t/test.c:712]:
```suggestion
    subtest("set_cc", test_set_cc);
```

**Review by kazuho — APPROVED** [review]
Thank you for the changes. LGTM.

**Pull request merged**